tests/native/scaled.cpp

```diff
@@ -1,14 +1,37 @@
+#include "gtest/gtest.h"
+
 #include <experimental/linalg>
 #include <experimental/mdspan>
+#include <type_traits>
 #include <vector>
-#include "gtest/gtest.h"
 
 namespace {
   using std::experimental::dynamic_extent;
   using std::experimental::extents;
   using std::experimental::mdspan;
   using std::experimental::linalg::scaled;
 
+  // scaled(1 << 20, scaled(1 << 20, x)) with x having value_type double
+  // should not overflow int.
+  TEST(scaled, preserve_original_evaluation_order)
+  {
+    constexpr double d = 4.2;
+    constexpr auto computed_val = (1 << 20) * ((1 << 20) * d);
+    static_assert(std::is_same_v<std::remove_cv_t<decltype(computed_val)>, double>);
+    constexpr auto expected_val = 1099511627776uLL * d;
+    static_assert(computed_val == expected_val);
+
+    double x_element = d;
+    mdspan<double, extents<1>> x(&x_element);
+    EXPECT_EQ(x[0], d);
+    auto x_scaled = scaled(1 << 20, x);
+    EXPECT_EQ(x_scaled[0], (1 << 20) * d);
+    auto x_scaled_scaled = scaled(1 << 20, x_scaled);
+    EXPECT_EQ(x_scaled_scaled[0], expected_val);
+    static_assert(std::is_same_v<decltype(x_scaled_scaled)::value_type, double>);
+    static_assert(std::is_same_v<decltype(x_scaled_scaled)::element_type, const double>);    
+  }
+
   TEST(scaled, mdspan_double_scalar_float)
   {
     using vector_element_type = double;
```

Comment:
There are compilation errors from `proxy_reference.hpp` when both `vector_element_type` and `scaling_factor_type` are of `std::complex<double>`.  I suppose this should work?

```
/home/yu/repos/nv/stdblas_test/stdBLAS/include/experimental/__p1673_bits/proxy_reference.hpp:163:46: error: no match for ‘operator+’ (operand types are ‘std::experimental::__p1673_version_0::linalg::impl::proxy_reference<std::complex<double>&, std::complex<double>, std::experimental::__p1673_version_0::linalg::scaled_scalar<std::complex<double>, std::complex<double>&, std::complex<double> > >::value_type’ {aka ‘std::complex<double>’} and ‘float’)
  163 |   P1673_PROXY_REFERENCE_ARITHMETIC_OPERATOR( + )
      |                                              ^
```

Reply:
@youyu3 Could that `float` perhaps come from `scaled` in a test?  The only thing that could be mixing `float` and `complex<double>` would be a test.

Reply:
Oh didn't read what the test was doing.  Yes you are right :).